internal/v4/archive.go

```diff
@@ -318,7 +335,7 @@ func (h *ReqHandler) addEntity(id *router.ResolvedURL, r io.ReadSeeker, blobName
 		}
 		if err := bundleData.VerifyWithCharms(verifyConstraints, charms); err != nil {
 			// TODO frankban: use multiError (defined in internal/router).
-			return errgo.Notef(verificationError(err), "bundle verification failed")
+			return errgo.NoteMask(verificationError(err), "bundle verification failed", errgo.Is(params.ErrInvalidEntity))
```

Comment:
an error with cause ErrInvalidEntity will be returned from verificationError if the error from VerifyWithCharms returns an error along those lines.